File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxi=nums[0]
        sm=0
        for i in range(len(nums)):
            sm+=nums[i]
            maxi=max(sm,maxi)
            if sm<0:
                sm=0
        return maxi

File: 0053-maximum-subarray/0053-maximum-subarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        int sm=nums[0];
        int maxi=nums[0];
       for(int i=1;i<nums.Length;i++){
        sm=Math.Max(nums[i],nums[i]+sm);
        maxi=Math.Max(sm,maxi);
       }
        return maxi;
    }
}